### PR TITLE
 Made LinkedList to work as java.util.LinkedList

### PR DESCRIPTION
**Pull request by SonyaPakhomova**

### CODE REVIEW

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 1]
```diff
@@ -1,47 +1,156 @@
...
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> previously;
```
Comment (sich-mykhailo) [inline src/main/java/core/basesyntax/MyLinkedList.java:14]: maybe better rename to previous?

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 2]
```diff
@@ -1,47 +1,156 @@
...
-        return null;
+        Node<T> node = getNode(index);
+        return removingLink(node);
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> node = head;
+        for (int i = 0; i < size; i++) {
+            if (Objects.equals(node.item, object)) {
```
Comment (sich-mykhailo) [inline src/main/java/core/basesyntax/MyLinkedList.java:87]: Don’t use class Objects.

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 3]
```diff
@@ -1,47 +1,156 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
```
Comment (sich-mykhailo) [inline src/main/java/core/basesyntax/MyLinkedList.java:4]: Don’t use class Objects.

**Review by sich-mykhailo — COMMENTED** [review]
checked

**Review by sich-mykhailo — APPROVED** [review]

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 4]
```diff
@@ -3,45 +3,153 @@
...
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("This index is out of bounds");
+        }
+        if (isEmpty() || index == size) {
+            add(value);
+            return;
+        }
+        Node<T> node = getNode(index);
```
Comment (Max-Pochepets) [inline src/main/java/core/basesyntax/MyLinkedList.java:44]:
```suggestion
        Node<T> nextNode = getNode(index);
```

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 5]
```diff
@@ -3,45 +3,153 @@
...
+            }
+        }
+        return node;
+    }
+
+    private T removingLink(Node<T> node) {
+        if (node.previous == null) {
+            if (size == 1) {
+                size--;
+                return node.item;
```
Comment (Max-Pochepets) [inline src/main/java/core/basesyntax/MyLinkedList.java:131]: will the node be stored in memory?

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

**Review by Max-Pochepets — APPROVED** [review]

**Pull request closed without merging**